0000-qiskit-pulse-schedule-relative-timing.md

```diff
@@ -52,49 +52,107 @@ Factors to consider:
- - max(|N<sub>C</sub>|) : the maximum number of instructions on any single channel within `self`
+ - N<sub>self</sub> : the total number of instructions in `self`
+ - N<sub>new</sub> : the total number of instructions in another `Schedule` being added in some way to `self`
+ - N<sub>C</sub> : the number of instructions on channel C within `self`
 
 ### New Schedule methods
 
 | Method        | Runtime        |  Description     |
 |---------------|----------------|------------------|
-|`barrier(self)`| O(\|C\|) | Add `Delay`s across `channels` so that all of the `channels` are occupied until `self.duration(channels)`. |
-|`left_align(self)` | O(\|C\|) | This is the default behavior. We could have it return `self`, or we could remove any pre-pended `Delay`s. The latter is probably better, because `schedule.right_align().left_align()` should yield the original `schedule`. Should we also append `Delay`s? Likely. *Needs more thought* |
-|`center_align(self)` | O(\|C\|) | Pre- and ap-pend equally sized `Delay`s (how to split odd durations?) on each channel so that all channels are occupied until `self.duration()` |
-|`right_align(self)` | O(\|C\|) | Prepend `Delay`s on each channel so that all channels are occupied until `self.duration()`. Should we also remove trailing `Delay`s? |
-
+|`barrier(self, channels=None)`| O(\|C\|) | Add `Delay`s across `channels` so that all of the `channels` are occupied until `self.duration(channels)`. By default, barrier on all channels. |
+|`left_align(self)` | O(\|C\|) | This is the default behavior. Return `self`.
+|`center_align(self)` | O(\|C\|) | Pre- and append equally sized `Delay`s (how to split odd durations?) on each channel so that all channels are occupied until `self.duration()` |
+|`right_align(self)` | O(\|C\|) | Prepend `Delay`s on each channel so that all channels are occupied until `self.duration()`.
+
+**Example**
+
+```
+sched = Schedule()
+sched += Delay(10, drive0)
+sched += Play(gaussian, drive0)
+sched = sched.barrier(drive0, drive1)
+subsched = Play([1.0]*20, drive1)
+subsched = Play([1.0]*10, drive0)
+sched += subsched.right_align()
+```
 
 ### API Changes
 
-\# TODO
-
 | Method        | Runtime        |  Change          |
 |---------------|----------------|------------------|
-| `append(self, schedule)` | O(\|N<sub>new</sub>\|) |  |
-|`insert(self, time, schedule)`| O(\|N<sub>new</sub>\| * \|N<sub>C</sub>\|) |  |
+| `append(self, schedule)` | O(\|C\|) | Previously, `append` would add `schedule` to `self` with relative times preserved, at time set by `self.duration({schedule.channels}.intersection({self.channels}))`. The new method will simply extend the instructions in `self` with those in `schedule` on a per channel basis. |
+|`insert(self, time, schedule)`| O(N<sub>new</sub> * N<sub>C</sub>) | This will be slow generally, but it can be implemented with the same runtime as `append` when the `time` is greater than `self.duration(schedule.channels)`. |
```

Comment:
Perhaps we should have both an `insert_at_time` and an `insert` (as the actual argument position in the schedule data structure) to properly differentiate these two operations?

Reply:
sounds good to me